jobs/deploy-webapp.groovy

```diff
@@ -615,25 +615,28 @@ def _monitor() {
       dir("webapp") {
          try {
             exec(cmd);
+            buildmaster.notifyMonitoringStatus(params.GIT_REVISION, "finished:succeeded");
          } catch (e) {
             sleep(1);   // give the watchdog a chance to notice an abort
             if (currentBuild.result == "ABORTED") {
                // Means that we aborted while running this, which
                // the watchdog (in vars/notify.groovy) noticed.
-               // We want to continue with the abort process.
+               // We want to continue with the abort process, while notifying buildmaster.
+               buildmaster.notifyMonitoringStatus(params.GIT_REVISION, "finished:aborted");
                throw e;
-            }
+            } else {
             // Failure to monitor is not a fatal error: we'll tell
             // people on slack so they can monitor manually.  But
             // we don't want to abort the deploy, like a FAILURE would.
+               buildmaster.notifyMonitoringStatus(params.GIT_REVISION, "finished:failed");
```

Comment:
Just so I'm clear - this means that the monitoring process couldn't complete due to an internal failure. not that the monitoring process detected elevated levels of errors, right?

Reply:
I believe you're right about this; I think there's a change to be made to monitor.py that should let it notify buildmaster directly when errors occur... or as Eric previously suggested, that buildmaster consume the monitoring directly. I think it would be easier to have buildmaster and monitor.py talk if the buildmaster token is present than it would be to replace monitor.py with buildmaster handling monitoring. Unless @yogieric can tell me that the monitor.py exits with a status other than `0` when errors are present, a cursory look at the monitor.py does not suggest to me that it does.